tests/io/test_api.py

```diff
@@ -31,15 +35,34 @@ def test_load_filter_nonchunked():
 
 @mark.api
 def test_load_monotonic():
-    data = aeon.load(monotonic_path, exp02.Patch2.Encoder)
+    data = aeon.load(monotonic_path, exp02.Patch2.Encoder, downsample=None)
     assert len(data) > 0 and data.index.is_monotonic_increasing
 
 
 @mark.api
 def test_load_nonmonotonic():
-    data = aeon.load(nonmonotonic_path, exp02.Patch2.Encoder)
+    data = aeon.load(nonmonotonic_path, exp02.Patch2.Encoder, downsample=None)
     assert not data.index.is_monotonic_increasing
 
 
+@mark.api
+def test_load_encoder_with_downsampling():
+    data = aeon.load(monotonic_path, exp02.Patch2.Encoder, downsample=True)
+    raw_data = aeon.load(monotonic_path, exp02.Patch2.Encoder, downsample=None)
+
+    # Check that the length of the downsampled data is less than the raw data
+    assert len(data) < len(raw_data)
+
+    # Check that the first timestamp of the downsampled data is within 20ms of the raw data
+    assert abs(data.index[0] - raw_data.index[0]).total_seconds() <= 0.02
+
+    # Check that the last timestamp of the downsampled data is within 20ms of the raw data
+    assert abs(data.index[-1] - raw_data.index[-1]).total_seconds() <= 0.02
+
+    # Check that the minimum difference between consecutive timestamps in the downsampled data
+    # is at least 20ms (50Hz)
+    assert data.index.to_series().diff().dt.total_seconds().min() >= 0.02
```

Comment:
By definition of downsampling, both the timestamps and the data points will be altered, so an equality test will never work, and besides the two sequences will have different lengths, again by construction. This code does not even compile since the `sort_values` function sorts on columns, and this addition will create all kinds of complexities in the code.

Note this is time-series data so the only relevant ordering is time, there is nothing in the ordering of the values which makes sense to assert here.